Mathlib/Combinatorics/SimpleGraph/Connectivity/Subgraph.lean

```diff
@@ -103,6 +103,18 @@ protected lemma Connected.sup {H K : G.Subgraph}
   rintro ⟨v, (hv|hv)⟩
   · exact Reachable.map (Subgraph.inclusion (le_sup_left : H ≤ H ⊔ K)) (hH ⟨u, hu⟩ ⟨v, hv⟩)
   · exact Reachable.map (Subgraph.inclusion (le_sup_right : K ≤ H ⊔ K)) (hK ⟨u, hu'⟩ ⟨v, hv⟩)
+
+lemma Connected.exists_connectedComponent_eq {H : Subgraph G}
+    (hc : H.Connected) (h : ∀ v ∈ H.verts, ∀ w, H.Adj v w ↔ G.Adj v w) :
+    ∃ c : G.ConnectedComponent, H.verts = c.supp := by
```

Comment:
Can't you get away with a much weaker assumption than `h`? Something like
```suggestion
    (hc : H.Connected) (h : G ≤ H.spanningCoe) :
    ∃ c : G.ConnectedComponent, H.verts = c.supp := by
```

Reply:
I changed this to be an implication, rather than an iff. Your suggestion implices that `H = ⊤`, so I don't think that's weaker. I can't see a more compact way to write this down due to the asymmetry between v being in H.verts and w not necessarily being, I also added a comment to stress this part:)

Reply:
Oh, right!